spec/datadog/di/transport_spec.rb

```diff
@@ -0,0 +1,96 @@
+require "datadog/di/spec_helper"
+require "datadog/di/transport"
+
+RSpec.describe Datadog::DI::Transport do
+  let(:agent_settings) do
+    double('agent settings')
```

Comment:
I think it makes sense to replace it with `instance_double` to avoid over stabbing or introducing non existing methods

Reply:
Done.